src/tests/performance/get_trans.rs

```diff
@@ -0,0 +1,237 @@
+
+        // Wait for correct state and account data.
+        wait_for_state(&node.rpc_url(), "proposing".into()).await;
+        let account_data =
+            wait_for_account_data(&node.rpc_url(), TEST_ACCOUNT, EXPECTED_RESULT_TIMEOUT)
+                .await
+                .expect("unable to get account data");
+
+        // Get transaction info by rpc to put in cache.
+        let tx = account_data.result.account_data.previous_transaction;
+        let _ = get_transaction_info(&node.rpc_url(), tx.clone())
+            .await
+            .expect("unable to get transaction info");
+
+        let mut tx_hash = [0u8; TX_HASH_LEN];
+        hex::decode_to_slice(&tx, &mut tx_hash as &mut [u8])
+            .expect("unable to decode transaction hash");
+
+        let mut synth_handles = Vec::with_capacity(synth_count);
+        let test_start = tokio::time::Instant::now();
+
+        for socket in synth_sockets {
+            synth_handles.push(tokio::spawn(simulate_peer(node_addr, socket, tx_hash)));
+        }
+
+        // wait for peers to complete
+        for handle in synth_handles {
+            let _ = handle.await;
+        }
+
+        let time_taken_secs = test_start.elapsed().as_secs_f64();
+
+        let snapshot = test_metrics.take_snapshot();
+        if let Some(latencies) = snapshot.construct_histogram(METRIC_LATENCY) {
+            if latencies.entries() >= 1 {
+                // add stats to table display
+                table.add_row(RequestStats::new(
+                    synth_count as u16,
+                    REQUESTS,
+                    latencies,
+                    time_taken_secs,
+                ));
+            }
+        }
+
+        node.stop().unwrap();
+    }
+
+    // Display results table
+    println!("\r\n{}", table);
+}
+
+async fn simulate_peer(node_addr: SocketAddr, socket: TcpSocket, tx_hash: [u8; TX_HASH_LEN]) {
+    let mut synth_node = SyntheticNode::new(&Default::default()).await;
+
+    // Establish peer connection
+    synth_node
+        .connect_from(node_addr, socket)
+        .await
+        .expect("unable to connect to node");
+
+    for seq in 0..REQUESTS {
+        let payload = Payload::TmGetObjectByHash(TmGetObjectByHash {
+            r#type: ObjectType::OtTransactions as i32,
+            query: true,
+            seq: Some(seq as u32),
+            ledger_hash: None,
+            fat: None,
+            objects: vec![TmIndexedObject {
+                hash: Some(tx_hash.into()),
+                node_id: None,
+                index: None,
+                data: None,
+                ledger_seq: None,
+            }],
+        });
+
+        // Query transaction via peer protocol.
+        if synth_node.is_connected(node_addr) {
+            synth_node
+                .unicast(node_addr, payload)
+                .expect("unable to send message");
+        } else {
+            synth_node.shut_down().await;
+            return;
+        }

```

Comment:
`else` block could be removed (and a simple break would lead to a shutdown and function return):
```rust
        // Query transaction via peer protocol.
        if !synth_node.is_connected(node_addr) {
            break;
        }

        synth_node
            .unicast(node_addr, payload)
            .expect("unable to send message");
```